apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PgNumeric.cs

```diff
@@ -0,0 +1,487 @@
+        }
+
+        /// <summary>
+        /// Implementation of <see cref="IComparable.CompareTo"/> to compare two <see cref="PgNumeric"/> values.
+        /// </summary>
+        /// <remarks>
+        /// This uses explicit interface implementation to avoid it being called accidentally. The generic implementation should usually be preferred.
+        /// </remarks>
+        /// <exception cref="ArgumentException"><paramref name="obj"/> is non-null but does not refer to an instance of <see cref="PgNumeric"/>.</exception>
+        /// <param name="obj">The object to compare this value with.</param>
+        /// <returns>The result of comparing this value with <paramref name="obj"/>. <paramref name="obj"/> is null, this method returns a value greater than 0.
+        /// </returns>
+        int IComparable.CompareTo(object obj)
+        {
+            if (obj == null)
+            {
+                return 1;
+            }
+            return obj is PgNumeric numeric
+                ? CompareTo(numeric)
+                : throw new ArgumentException($"Object must be of type {nameof(PgNumeric)}", nameof(obj));
+        }
+
+        /// <summary>
+        /// Compares this value with <paramref name="other"/> for equality.
+        /// </summary>
+        /// <param name="other">The value to compare with this one.</param>
+        /// <returns><c>true</c> if <paramref name="other"/> is an equal <see cref="PgNumeric"/> value; <c>false</c> otherwise.</returns>
+        public bool Equals(PgNumeric other)
+        {
+            if (_notANumber || other._notANumber) // If any or both of the numeric value are NaN, they can't be equal.  
+            {
+                return false;
+            }
+
+            // if scale and integer value are same and both are not NaN, return true, else check if normalized values are equal.
+            if (_integer == other._integer && _scale == other._scale)
+            {
+                return true;
+            }
+
+            // 1 == 1.0 == 1.00 == 1.000, so trailing zeros if any must be normalized.
+            // If we are here, we have 2 numbers to be compared which may have trailing zeros, so check if normalized values are equal.
+            return Normalize(this).NormalizedValue == Normalize(other).NormalizedValue;
+        }
+
+        /// <summary>
+        /// Removes trailing zeros on the value and returns normalized value and scale.
+        /// Normalized value of 1.000, 1.00, 1.0, 1 is 1 and Normalized scale is 3, 2, 1, 0 respectively.
+        /// Noemalized value of 10.000, 10.00, 10.0, 10 is 10 and Normalized scale is 3, 2, 1, 0 respectively.
+        /// </summary>
+        /// <param name="pgNumeric">The PgNumeric value to be normalized.</param>
+        /// <returns>The normalized value and scale.</returns>
+        private static (BigInteger NormalizedValue, int NormalizedScale) Normalize(PgNumeric pgNumeric)
```

Comment:
If this remains useful after review, it should be called TrimTrailingZeros or something similar.